clash-ghc/src-ghc/Clash/GHC/ClashFlags.hs

```diff
@@ -95,6 +95,7 @@ flagsClash r = [
   , defFlag "fclash-edalize"                     $ NoArg (liftEwM (setEdalize r))
   , defFlag "fclash-no-render-enums"             $ NoArg (liftEwM (setNoRenderEnums r))
   , defFlag "fclash-timescale-precision"         $ SepArg (setTimescalePrecision r)
+  , defFlag "fclash-sanitize-netlist-ids"        $ NoArg (liftEwM (setSanitizeNames r))
```

Comment:
New flags should be documented in [this file](https://github.com/clash-lang/clash-compiler/blob/master/docs/developing-hardware/flags.rst). I would make sure we document

  - that some parts of vendor tooling may not accept all valid basic identifiers
  - this flag may result in user-specified names being mangled
  - this flag is a bit of a big hammer and shouldn't be used unless needed